File: TDDMicroExercises.YoursSolutions/Sam-Serpoosh/code/tire_pressure_monitoring_system/alarm.rb
require_relative "./sensor"

class Alarm
  LOWPRESSURETHRESHOLD = 17
  HIGHPRESSURETHRESHOLD = 21

  attr_reader :alarm_count

  def initialize(sensor=Sensor.new)
    @sensor = sensor 
    turn_off_alarm
    @alarm_count = 0
  end

  def check 
    psi_pressure = @sensor.pop_next_pressure_psi_value

    if dangerous_pressure?(psi_pressure)
      turn_on_alarm
      @alarm_count += 1
    end
  end

  def on? 
    @alarm_on
  end

  private
  def dangerous_pressure?(pressure)
    pressure < LOWPRESSURETHRESHOLD || 
      pressure > HIGHPRESSURETHRESHOLD
  end

  def turn_on_alarm
    @alarm_on = true
  end

  def turn_off_alarm
    @alarm_on = false
  end
end

File: TDDMicroExercises.YoursSolutions/Java-benwu/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm
{
    private final double LowPressureThreshold = 17;
    private final double HighPressureThreshold = 21;

    private Sensor sensor = new Sensor();

    private boolean alarmOn = false;

    public Alarm(Sensor sensor) {
        this.sensor = sensor;
    }

    public Alarm() {

    }

    public void check()
    {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LowPressureThreshold || HighPressureThreshold < psiPressureValue)
        {
            alarmOn = true;
        }
    }

    public boolean isAlarmOn()
    {
        return alarmOn; 
    }
}
